libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigationProvider.kt

```diff
@@ -0,0 +1,57 @@
+package com.mapbox.navigation.core
+
+import android.content.Context
+import com.mapbox.android.core.location.LocationEngine
+import com.mapbox.android.core.location.LocationEngineProvider
+import com.mapbox.android.core.location.LocationEngineRequest
+import com.mapbox.navigation.base.options.NavigationOptions
+
+
+object MapboxNavigationProvider {
+    @Volatile
+    private var mapboxNavigation: MapboxNavigation? = null
+
+    /**
+     *  Pass NavigationInitializationBundle to initialize a new instance of MapboxNavigation
+     *  Old instance will be cleaned up, if exists.
+     *  Don't pass any params to get current MapboxNavigation instance
+     */
+    fun navigation(bundle: NavigationInitializationBundle? = null): MapboxNavigation {
+        var instance = mapboxNavigation
+        if (bundle == null) {
+            if (instance == null) {
+                synchronized(MapboxNavigationProvider::class.java) {
+                    instance = mapboxNavigation
+                    if (instance == null) {
+                        throw RuntimeException("Provide options to init MapboxNavigation before using it.")
+                    } else {
+                        return instance!!
+                    }
+                }
+            } else {
+                return instance!!
+            }
+        } else {
+            synchronized(MapboxNavigationProvider::class.java) {
+                mapboxNavigation?.onDestroy()
+                mapboxNavigation = MapboxNavigation(
+                    bundle.context,
+                    bundle.navigationOptions,
+                    bundle.locationEngine,
+                    bundle.locationEngineRequest
+                )
+
+                return mapboxNavigation!!
+            }
+        }
+    }
+}
+
+data class NavigationInitializationBundle(
```

Comment:
```suggestion
data class NavigationInitializationOptions(
```

Unclear what the difference is between `Bundle` and `Options`. Mind if we stick to one word?

Reply:
ref: the longer term painful change is to make it so we only have NavigationOptions

https://github.com/mapbox/mapbox-navigation-android/pull/3191

Reply:
done